addons/account/models/account_bank_statement.py

```diff
@@ -683,7 +683,7 @@ def _prepare_counterpart_move_line_vals(self, counterpart_vals, move_line=None):
             if balance_to_reconcile is None else balance_to_reconcile
 
         if 'currency_id' in counterpart_vals:
-            currency_id = counterpart_vals['currency_id'] or company_currency.id
+            currency_id = counterpart_vals['currency_id'] or foreign_currency.id
```

Comment:
This must be fixed in 14.0.

Reply:
Hello Laurent, I rebased this PR to 14.0